### PR TITLE
feat(BE-70):Set default Profile picture to Generic Avatar

### PR DESCRIPTION
**Pull request by noornee**
**Linear Ticket:** 
- https://linear.app/team-descripto/issue/BE-70/set-default-profile-picture-to-generic-avatar

**Details**
 - checks if a bucket key "default" exists in s3 bucket
 - creates a "default" key if it doesnt exist and uploads it to s3 bucket
 - during user signup. set up a default profile/avatar for the user

**Preview:**
2022-12-02-233608_1733x381_scrot

2022-12-02-233652_887x364_scrot

### CODE REVIEW

File: pkg/repository/storage/s3/s3upload.go [hunk 1]
```diff
@@ -64,3 +70,48 @@ func UploadImage(file io.ReadCloser, filename string) (string, error) {
...
+	svc := s3.New(s3session)
+	MyBucket = config.GetConfig().S3.BucketName
+
+	_, err := svc.HeadObject(&s3.HeadObjectInput{
+		Bucket: aws.String(MyBucket),
+		Key:    aws.String(key),
+	})
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
```
Comment (kataszn) [inline pkg/repository/storage/s3/s3upload.go:85]: you should use **err** always for any error
no need to have these weird variable names
once an **err** is handled is should be used for other errors

File: pkg/repository/storage/s3/s3upload.go [hunk 2]
```diff
@@ -64,3 +70,48 @@ func UploadImage(file io.ReadCloser, filename string) (string, error) {
...
+
+// This sets up an s3 bucket key named "default" for the client
+// it also sets up a default profile picture for new users.
+func DefaultProfile() (profile_url, profile_key string) {
+	//func DefaultProfile() (string, string) {
+	cwd, _ := os.Getwd()
+	file, _ := os.Open(fmt.Sprintf("%s/static/avatar.jpg", cwd))
+	defer file.Close()
+
+	profile_url = fmt.Sprintf("https://miner-pictures.s3.amazonaws.com/%s", constants.S3_generic_avatar_key)
```
Comment (kataszn) [inline pkg/repository/storage/s3/s3upload.go:105]: it would've been better to get the url from a .env or config file

**Review by kataszn — APPROVED** [review]
Overall LGTM

File: pkg/repository/storage/s3/s3upload.go (hunk 1 above)
Comment (noornee) [reply pkg/repository/storage/s3/s3upload.go:85]: okay, my bad :sweat:

File: pkg/repository/storage/s3/s3upload.go (hunk 2 above)
Comment (noornee) [reply pkg/repository/storage/s3/s3upload.go:105]: i was thinking of doing that. can i still make changes to this now?

**Pull request merged**